### PR TITLE
dockerprune: only prune images if they match a running manifest

### PR DESCRIPTION
**Pull request by maiamcc**
it was a pain to run tilt.build b/c the docker pruner on a servantes run had pruned away
all my tilt.build images/caches

with this PR, the docker pruner only prunes images matching refs currently being run.

I BELIEVE this will mean that we only prune docker caches corresponding to those images -- b/c we're not pruning dangling caches, so we shouldn't delete any that relate to extant images? lmk if this is wrong.

for future PR: apply this behavior to `tilt docker-prune` so it gives an accurate preview

### CODE REVIEW

File: internal/cli/docker_prune.go [hunk 1]
```diff
@@ -14,6 +17,18 @@ type dockerPruneCmd struct {
 	maxAgeStr string
 }
 
+type dpDeps struct {
```
Comment (jazzdan) [inline internal/cli/docker_prune.go]: This type and the constructor aren't used?

File: internal/engine/dockerprune/docker_pruner.go [hunk 2]
```diff
@@ -60,6 +62,14 @@ func (dp *DockerPruner) OnChange(ctx context.Context, st store.RStore) {
 	curBuildCount := state.CompletedBuildCount
 	hasDockerBuild := state.HasDockerBuild()
 	nextToBuild := buildcontrol.NextManifestNameToBuild(state)
+	var imgSelectors []container.RefSelector
+	for _, m := range state.Manifests() {
+		for _, iTarg := range m.ImageTargets {
+			sel := iTarg.ConfigurationRef
+			sel.AsNamedOnly()
```
Comment (nicks) [inline internal/engine/dockerprune/docker_pruner.go]: this is a no-op?

File: internal/engine/dockerprune/docker_pruner.go [hunk 3]
```diff
@@ -60,6 +62,14 @@ func (dp *DockerPruner) OnChange(ctx context.Context, st store.RStore) {
 	curBuildCount := state.CompletedBuildCount
 	hasDockerBuild := state.HasDockerBuild()
 	nextToBuild := buildcontrol.NextManifestNameToBuild(state)
+	var imgSelectors []container.RefSelector
+	for _, m := range state.Manifests() {
+		for _, iTarg := range m.ImageTargets {
+			sel := iTarg.ConfigurationRef
```
Comment (nicks) [inline internal/engine/dockerprune/docker_pruner.go]: i'm pretty sure this should be DeploymentRef, not ConfigurationRef?

DeploymentRef is what it actually gets tagged with

File: internal/cli/docker_prune.go (hunk 1 above)
Comment (maiamcc) [reply internal/cli/docker_prune.go]: lol ya they belong in a different PR, just removed them

File: internal/engine/dockerprune/docker_pruner.go (hunk 3 above)
Comment (maiamcc) [reply internal/engine/dockerprune/docker_pruner.go]: oh yep you're right.

File: internal/engine/dockerprune/docker_pruner.go (hunk 2 above)
Comment (maiamcc) [reply internal/engine/dockerprune/docker_pruner.go]: nope, `ConfigurationRef` is actually of type `RefSelector` 😒 but it's irrelevant now that i'm using `DeploymentRef`!

**Pull request merged**